### PR TITLE
chore(cli): ensure new workloads are not named "pipelines"

### PR DESCRIPTION
**Pull request by dannyrandall**
With the new pipelines file structure, if a new workload is named `pipelines`, it will be potentially confusing with the copilot directory for pipelines. To avoid that issue, this change will ensure new workloads do not use that name.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the Apache 2.0 License.

### CODE REVIEW

File: internal/pkg/cli/validate.go [hunk 1]
```diff
@@ -161,6 +162,14 @@ var (
 
 const regexpFindAllMatches = -1
 
+// reservedWorkloadNames is a constant map of reserved workload names that users are not allowed to name their workloads
+func reservedWorkloadNames() map[string]bool {
+	return map[string]bool{
+		"pipelines":    true, // reserved to avoid directory conflict with copilot pipelines
+		"environments": true, // reserved to avoid directory conflict with copilot enviornments
```
Comment (efekarakus) [inline internal/pkg/cli/validate.go]: 😍 
```suggestion
		"environments": true, // reserved to avoid directory conflict with copilot environments
```

**Review by efekarakus — APPROVED** [review]

File: internal/pkg/cli/validate_test.go [hunk 2]
```diff
@@ -942,3 +947,30 @@ func Test_validateSubscribe(t *testing.T) {
 		})
 	}
 }
+
+func TestValidateJobName(t *testing.T) {
```
Comment (huanjani) [inline internal/pkg/cli/validate_test.go:951]: thanks for adding these other ones, too!

File: internal/pkg/cli/validate.go [hunk 3]
```diff
@@ -179,11 +188,27 @@ func validateSvcName(val interface{}, svcType string) error {
...
 	return nil
 }
 
-func validateSvcPort(val interface{}) error {
+func validateNotReservedWorkloadName(val interface{}) error {
+	name, ok := val.(string)
+	switch {
+	case !ok:
+		return errValueNotAString
+	case reservedWorkloadNames()[name]:
```
Comment (huanjani) [inline internal/pkg/cli/validate.go:204]: this is super cool.

**Review by huanjani — APPROVED** [review]
🚢

**Pull request merged**